### ISSUE TITLE
Windows 10 Got socket error trying to find package

### ISSUE THREAD

**Issue by samueladekunle** [flutter/flutter#22474]
After creating a new flutter project, and running `flutter packages get`  in the command line, what I see is
```
Running "flutter packages get" in startup_namer...
Got socket error trying to find package cupertino_icons at https://pub.dartlang.org.
pub get failed (69) -- attempting retry 1 in 1 second...
```
and on running `flutter doctor -v`, this is the output
```
[√] Flutter (Channel beta, v0.8.2, on Microsoft Windows [Version 10.0.17134.286], locale en-US)
    • Flutter version 0.8.2 at C:\Flutter\flutter
    • Framework revision 5ab9e70727 (3 weeks ago), 2018-09-07 12:33:05 -0700
    • Engine revision 58a1894a1c
    • Dart version 2.1.0-dev.3.1.flutter-760a9690c2

[√] Android toolchain - develop for Android devices (Android SDK 28.0.0)
    • Android SDK at C:\Users\AUNTYHANNAH\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.0
    • ANDROID_HOME = C:\Users\AUNTYHANNAH\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)
    • All Android licenses accepted.

[√] Android Studio (version 3.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b02)

[√] VS Code (version 1.27.2)
    • VS Code at C:\Users\AUNTYHANNAH\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 2.18.0

[√] VS Code, 64-bit edition (version 1.17.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 2.18.0

[!] Connected devices
    ! No devices available

! Doctor found issues in 1 category.

```
The `flutter packages get` command worked perfectly a couple of hours ago in another project's directory.
Am not behind any proxy server, neither am I located in China. Thanks in advance.

**Comment by zoechi** [comment]
Either a network issue or a temprorary fluke on pub.dartlang.org.

**Comment by samueladekunle** [comment]
@zoechi Thanks for your response, I don't think it is a network issue, have kept on trying this for hours now, still keep getting the same errors.

**Comment by samueladekunle** [comment]
The Flutter SDK was unzipped in C:/Flutter, where I actually have R/W access to, since am logged in as the admin. What I have actually noticed is that I cannot access `https://pub.dartlang.org` from my browser, but I can for `https://storage.googleapis.com`

**Comment by samueladekunle** [comment]
It seems the server at https://pub.dartlang.org/ has blocked my IP Address, `ping`ing `https://pub.dartlang.org` returns `Request timed out` and that seems strange from a package repository.

**Comment by zoechi** [comment]
I doubt pub.dartlang.org does something like that.

**Comment by zoechi** [comment]
This looks like a network issue on your site.

**Comment by samueladekunle** [comment]
Not a network issue at all. In fact not until I went through a VPN connection before I was able to download packages from pub.dartlang.org. Presently, each time I want to get a package from pub.dartlang.org, I've to go through a VPN server.

**Comment by zoechi** [comment]
@samueladekunle 

>  I've to go through a VPN server.

I'd say that's a strong indication that it actually is a network issue on your end.

Are you in China?

**Comment by samueladekunle** [comment]
Not at all, am in Nigeria

**Comment by samueladekunle** [comment]
A fellow Nigerian seems to have faced a similar issue as posted [here](https://stackoverflow.com/questions/52158282/got-socket-error-trying-to-find-package-cupertino-icons-at-https-pub-dartlang)

**Comment by zoechi** [comment]
I think this should be reported in https://github.com/dart-lang/pub/issues/new

**Comment by samueladekunle** [comment]
@zoechi I just did that [here](https://github.com/dart-lang/pub/issues/1989). Thanks a lot.

**Comment by zoechi** [comment]
Thanks!
Closing here

**Comment by Emmanuelasika** [comment]
> 
> 
> A fellow Nigerian seems to have faced a similar issue as posted [here](https://stackoverflow.com/questions/52158282/got-socket-error-trying-to-find-package-cupertino-icons-at-https-pub-dartlang)

am also facing this

**Comment by samueladekunle** [comment]
@Emmanuelasika try using a VPN service or a public Wi-Fi. You can also try one of the [download mirrors meant for China](https://flutter.io/community/china),

**Comment by ayodelesalimon** [comment]
Having same issues. Am in Nigeria

**Comment by zoechi** [comment]
https://www.dartlang.org/tools/pub/troubleshoot#pub-get-fails-from-behind-a-corporate-firewall might help some

**Comment by saijayaprakash** [comment]
I am having the same problem 

> ### **'Got Socket Error trying to get vector_math at ...'**

I am able to open "https://pub.dartlang.org/" in browser without any proxy.

**Comment by royken** [comment]
@samueladekunle have you found any workaround for this issue ? I am having the same problem here, on my Windows 10 OS computer, but it works well on ubuntu OS computer. Both computers are on the same network

**Comment by hashweather** [comment]
I am also still having this same issue as well. As a matter of fact, I can't move any further in the project as I can't get the necessary plugins I need to work with from pub.dartlang.org. Being stalled for days because of this. So frustrating and discouraging. I am running Ubuntu OS and working under a very strong network, NO VPN no PROXY just USB Mobile Tethering.

**Comment by Emmanuelasika** [comment]
> I am also still having this same issue as well. As a matter of fact, I can't move any further in the project as I can't get the necessary plugins I need to work with from pub.dartlang.org. Being stalled for days because of this. So frustrating and discouraging. I am running Ubuntu OS and working under a very strong network, NO VPN no PROXY just Mobile USB Tethering.

Hello hash, i'd suggest you download and install a VPN on your system.
i had the same issue until i started using VPN.. the free one i use is called AVIRA VPN

**Comment by kels-orien** [comment]
I'm also in Nigeria. Having same issues

**Comment by royken** [comment]
Hello, my problem was related to Kaspersky antivirus. Once I unsintalled and reinstalled it, everything worked great

**Comment by fioan89** [comment]
Same situation here. Recently I was trying to setup flutter. Turn's out I did not set-up the proxy correctly, upper case env variables are not honored, nor hostnames  with protocol included in the value.
If you are also behind a proxy then please make sure you strictly follow the format from https://dart.dev/tools/pub/troubleshoot#pub-get-fails-from-behind-a-corporate-firewall

If you are on Windows and you run the flutter/pub command from PowerShell, the usual recommendation set https_proxy=proxy.yourproxy.com:8080 will not work. The correct ways is:
$env:https_proxy='proxy.yourproxy.com:8080'

**Comment by shajji1** [comment]
can anyone tell me how to solve this problem in windows 10?

**Comment by sheanWinston** [comment]
Please if you are in Nigeria try not to use glo to access the dartlang site. if the issue persists after a change of network....

**Comment by shajji1** [comment]
@sheanWinston, i'm not in nigeria, and i'm getting **Got TLS error trying to find package cupertino_icons at https://pub.dartlang.org.** while running command **flutter packages get**, is there any way to change in dart files, so it can accept bad certificates.

**Comment by sheanWinston** [comment]
@shajji1 Try accessing the site https://pub.dartlang.org
With your browser

**Comment by shajji1** [comment]
i checked it, but it routes me to this url https://pub.dev/,

**Comment by sheanWinston** [comment]
Hmm. Got no other idea

**Comment by shajji1** [comment]
@sheanWinston i also run this command **flutter packages get -v** and getting error while accessing this url **https://pub.dartlang.org/api/packages/cupertino_icons**, but when i access it via browser it works smoothly.

**Comment by shajji1** [comment]
In flutter is there any config like in **npm** `strict-ssl `to accept bad certificates? sorry for bothering you.

**Comment by sheanWinston** [comment]
Not so sure, but you can Google it

On Sep 30, 2019 8:24 AM, "shajji1" <notifications@github.com> wrote:

@sheanWinston <https://github.com/sheanWinston> In flutter is there any
config like in *npm* strict-ssl to accept bad certificates? sorry for
bothering you.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/flutter/flutter/issues/22474?email_source=notifications&email_token=AH5FUI3NS5ZT5YUYHJEAWSDQMGSUNA5CNFSM4FYCXRX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD74WE3Q#issuecomment-536437358>,
or mute the thread

**Comment by shajji1** [comment]
k, thank you for your time. :)

**Comment by sheanWinston** [comment]
U re welcome

On Sep 30, 2019 8:41 AM, "shajji1" <notifications@github.com> wrote:

@sheanWinston <https://github.com/sheanWinston> k, thank you for your time.
:)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/flutter/flutter/issues/22474?email_source=notifications&email_token=AH5FUI7LUJJRHK5MB26RJSTQMGUT5A5CNFSM4FYCXRX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD74XRPQ#issuecomment-536443070>,
or mute the thread

**Comment by FuturetheCoder** [comment]
Got socket error trying to find package flutter_web_view at https://pub.dartlang.org.
pls how can i fix this

**Comment by FuturetheCoder** [comment]
guys i just fixed my error  (DO NOT USE GLO WHEN ACCESSING PUB.DARTLANG.ORG),

**Comment by Greatcallie** [comment]
I also have the same issue. And it goes away sometimes and keeps returning. And yes am in Nigeria

**Comment by sheanWinston** [comment]
Try to use a fast network. That command runs a request and downloads a
large file. So make sure to use a network provider that is fast in your
region

**Comment by Greatcallie** [comment]
No it's not network speed. It's an IP block issue

**Comment by Greatcallie** [comment]
These DNS on my Mac works. 198.153.192.40 and 198.153.194.40

**Comment by atulaa1** [comment]
I have same issue , when i try access https://pub.dev/ in my browser . It redirectly to https://pub.dev/
I also try to set http-proxy. But it still not working.
I come from Vietnam. 
This is my first flutter app. But i cannot run it now =))

**Comment by jpainam** [comment]
This from SO solved the issue

https://stackoverflow.com/questions/52158282/got-socket-error-trying-to-find-package-cupertino-icons-at-https-pub-dartlang

I would advice anyone that might still be having this problem to TRY the following simply steps: 1) Open environment variable settings 2)create a new variable with the name: PUB_HOSTED_URL and its value: https://pub.flutter-io.cn 3)create a new variable with the name: FLUTTER_STORAGE_BASE_URL and its value: https://storage.flutter-io.cn

**Comment by Tarhex** [comment]
Try to see if you can reach https://dartlang-pub-dev.appspot.com/ in your browser

If it is accessible, declare PUB_HOSTED_URL as a system environment variable and give it the value https://dartlang-pub-dev.appspot.com/

Then exit and reopen your cmd window and try again.

**Comment by mrpbboss** [comment]
If your living in Nigeria use MTN network, because it's supports ipv4.If your using Google network you will get this error your getting, because it uses Ipv6 and flutter supports ipv4.@zoechi

**Comment by Omzlaw** [comment]
For GLO users in Nigeria, you can change your access point name to glosecure username: gprs and password: gprs

**Comment by mrpbboss** [comment]
I meant Glo network

**Comment by DrNerd97** [comment]
@Omzlaw, This doesn't work... I had to download Opera that comes preinstalled with a VPN, and it seemed OK for downloading some packages, but when I try to clone a Flutter codebase, I still get Got socket error trying to find package cupertino_icons at https://pub.dartlang.org

**Comment by promzybankz** [comment]
> Not at all, am in Nigeria

I'm In Nigerial also and I'm Seriously facing this issue

**Comment by ayodelesalimon** [comment]
Download a VPN and use.

**Comment by promzybankz** [comment]
> I'm also in Nigeria. Having same issues.Thanks for the solution

If you are in Nigeria... GLO is the problem. use a VPN or change to MTN or Airtel.... chikina problem solved ini 4.7s

**Comment by DrNerd97** [comment]
[Update]: NordVPN did the trick
